test/test_requests/test_patch.py

```diff
@@ -0,0 +1,64 @@
+import os
+
+import requests
+
+from .utils import setup_env_vars
+
+
+from kw.platform.requests import (  # isort:skip
+    construct_user_agent,
+    patch,
+    patch_with_user_agent,
+)
+
+
+def test_construct_user_agent():
+    user_agent = construct_user_agent()
+    assert user_agent == "unittest/1.0 (Kiwi.com test-env)"
+
+    user_agent = construct_user_agent(
+        app_name="different-test",
+        package_version="2.0",
+        app_environment="different-test-env",
+    )
+    assert user_agent == "different-test/2.0 (Kiwi.com different-test-env)"
+
+
+def test_construct_user_agent_missing_args():
+    # Cleanup env vars.
+    # Otherwise the 'None' arguments would get overwritten
+    os.environ["APP_NAME"] = ""
+    os.environ["PACKAGE_VERSION"] = ""
+    os.environ["APP_ENVIRONMENT"] = ""
+
+    user_agent_1 = construct_user_agent(
+        app_name=None, package_version="2.0", app_environment="different-test-env"
+    )
+    user_agent_2 = construct_user_agent(
+        app_name="different-test",
+        package_version=None,
+        app_environment="different-test-env",
+    )
+    user_agent_3 = construct_user_agent(
+        app_name="different-test", package_version="2.0", app_environment=None
+    )
+    for user_agent in (user_agent_1, user_agent_2, user_agent_3):
+        assert user_agent is None
+
+    # Fix the env vars
+    setup_env_vars()
+
+
+def test_patch_with_user_agent():
+    res = requests.get("http://kiwi.com")
+    assert res.request.headers["User-Agent"].startswith("python-requests")
+
+    patch_with_user_agent()
```

Comment:
The problem with this is that it will patch requests globally. So in all other tests, requests are patched. It will be a problem if, for example, you use `pytest.mark.parametrize` and run the test several times with different parameters - the test will pass only in the first run, in all the following runs the requests are already patched.